lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +229,31 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        url = ""
+        links = article_bs.find_all('a', class_="article-item_title")
+        for link in links:
+            print(link)
+            url = link.get('href')
+            print(url)
+            if url not in self.urls:
+                break
+        url = self.url_pattern + url[len("/articles")::]
+        return url
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        seed_urls = self.get_search_urls()
+
+        for seed_url in seed_urls[:1]:
+            response = make_request(seed_url, self.config)
+            if not response.ok:
+                continue
+
+            article_bs = BeautifulSoup(response.text, "html.parser")
+            urls = [self._extract_url(article_bs) for _ in range(10)]
```

Comment:
setting your loop to exact range 10 here is not safe

are you sure this number of iterations is enough in all crawler uasage cases? e.g. i would like to crawl 100 articles?

Reply:
what about using while?

Reply:
done